packages/mantine/src/components/table/table-edit-columns-visibility/TableEditColumnsVisibility.tsx

```diff
@@ -0,0 +1,91 @@
+import {Button, Checkbox, Divider, Grid, Popover, ScrollArea, Stack} from '@mantine/core';
+import {FunctionComponent} from 'react';
+
+import {TableComponentsOrder} from '../Table.styles';
+import {useTable} from '../TableContext';
+import useStyles from './TableEditColumnsVisibility.styles';
+import {TableEditColumnsVisibilityProps} from './TableEditColumnsVisibility.types';
+
+const COLUMNS_IDS_TO_EXCLUDE = ['collapsible', 'select'];
+
+export const TableEditColumnsVisibility: FunctionComponent<TableEditColumnsVisibilityProps> = ({
+    classNames,
+    styles,
+    unstyled,
+    label = 'Edit columns',
+    buttonVariant = 'outline',
+    showVisibleCountLabel = false,
+    nonHideableColumns,
+    maxSelectableColumns,
+    columnNames,
+}) => {
+    const {classes} = useStyles(null, {name: 'TableEditColumnsVisibility', classNames, styles, unstyled});
+    const {getAllColumns} = useTable();
+
+    const filteredColumns = getAllColumns().filter((column) => !COLUMNS_IDS_TO_EXCLUDE.includes(column.id));
+
+    const selectedColumnsCount = filteredColumns.filter((column) => column.getIsVisible()).length;
+    const adjustedMaxSelectableColumns = maxSelectableColumns
+        ? Math.min(maxSelectableColumns, filteredColumns.length)
+        : filteredColumns.length;
+
+    const remainingColumnsCount = adjustedMaxSelectableColumns
+        ? adjustedMaxSelectableColumns - selectedColumnsCount
+        : filteredColumns.filter((column) => !column.getIsVisible() && !nonHideableColumns?.includes(column.id)).length;
+
+    let footerText;
+    if (selectedColumnsCount === filteredColumns.length) {
+        footerText = 'All available columns selected';
+    } else if (adjustedMaxSelectableColumns && selectedColumnsCount >= adjustedMaxSelectableColumns) {
+        footerText = 'Maximum columns selected';
+    } else {
+        footerText = `You can choose ${remainingColumnsCount} more`;
+    }
```

Comment:
This footerText should be created using props, this way it can be overwritten

Reply:
How can I deal with string from a props but still have the count here? 🤔  

`remaningColumnsCount` is calculated at line 34 and is not accessible outside the component. 👀 

<img width="620" alt="image" src="https://github.com/coveo/plasma/assets/63734941/e22394a3-5287-419f-a4f3-f03eb8db08da">
****

Reply:
I got a new mockup for the footer so I think I will maybe juste put a `footer: reactNode` in the prop instead? 🤔 

I can also keep the current logic for the default footer if you think it can be used in the future